server/controllers/AuthController.js

```diff
@@ -157,6 +173,44 @@ class AuthController {
         return done(null, user);
       });
   }
-}
 
-export default AuthController;
+  /**
+   * @description Handles logging in a user locally
+   * @param {Object} req The HTTP request payload object
+   * @param {Object} res The HTTP response payload object
+   */
+  static login(req, res) {
+    const { email, username, password } = req.body;
+    User.findOne({
+      where: { $or: [{ email }, { username }] },
+    })
+      .then((user) => {
+        if (!user) {
+          const errors = { user: ['No user with the provided credentials found'] };
+          return AuthController.failureResponse({ code: 404, errors }, res);
+        }
+
+        if (!user.verified) {
+          const errors = { user: ['You must verify your email before you can login!'] };
+          return AuthController.failureResponse({ code: 403, errors }, res);
+        }
+
+        const passwordIsValid = bcrypt.compareSync(password, user.password);
+        if (!passwordIsValid) {
+          const errors = { user: ['Incorrect credentials! Please try again'] };
+          return AuthController.failureResponse({ code: 401, errors }, res);
+        }
+
+        const { id, firstName, lastName } = user;
+
+        const status = { code: 200, message: 'login successful!' };
+        AuthController.successResponse(status, {
```

Comment:
Well done on completing this task @Veraclins. I love the fact that you took initiative and tried to optimize the codebase by creating to methods to handle successful and failed authentication. It shows that you are really attentive to details. However, on this line, you added the user's email and password to the user object you sent to the `successResponse` method. I believe the method signs the token, but then are you certain that the said user did log in with a username in lieu of an email? Please correct me if I'm mistaken.

Reply:
Thank @chukwuemekachm for your feedback. I don't quite get what you mean by adding the password to the user object sent to the `succesResponse` method. As for whether or not the user signed in with username or email, I don't think that matters as the response on successful login should be the same for all users irrespective of how they login.

Reply:
Sorry I made a mistake, I meant email and username. However, I just corrected that.
What I meant is as follows:
- You destructured `email`, `username` and `password` from the request body. It can only be one of either `username` and `email`. I hoped you would reassign them from the user's object on successful login.
- Now if the user logs in with a `username` the `email` would be null and vice-versa.
- You sent those variables along with the user object. Please resolve it.

Reply:
Thanks for that. I now understand your point I have made the required changes. Thanks for pointing it out!